src/transformers/models/whisper/generation_whisper.py

```diff
@@ -567,11 +567,14 @@ def generate(
         # 3. Retrieve logits processors
         device = kwargs["encoder_outputs"][0].device if "encoder_outputs" in kwargs else input_features.device
         begin_index = init_tokens.shape[1]
+        num_beams = (
+            generation_config.num_beams if generation_config.num_beams is not None else kwargs.get("num_beams", 1)
+        )
```

Comment:
If I remember correctly, the kwargs arguments should have priority over the generation config, which is not the case here.

cc @ArthurZucker could you confirm ?

Reply:
yep! 
```suggestion
        num_beams = kwargs.get("num_beams", generation_config.get("num_beams", 1))
        )
```
rather no? (default to kwargs if wargs is not None